app/routes/members-area.my-account/route.tsx

```diff
@@ -23,36 +22,34 @@ export async function loader({ request }: LoaderFunctionArgs) {
 export default function MyAccountPage() {
     const { user } = useLoaderData<typeof loader>();
 
-    const [discardConfirmationOpen, setDiscardConfirmationOpen] = useState(false);
-
-    const {
-        firstName,
-        setFirstName,
-        lastName,
-        setLastName,
-        phone,
-        setPhone,
-        isUpdating,
-        isResettingPassword,
-        updateAccountDetails,
-        discardChanges,
-        sendPasswordResetEmail,
-    } = useAccountDetailsForm(user?._id, {
+    const initialUserDetailsFormData = {
         firstName: user?.contact?.firstName ?? '',
         lastName: user?.contact?.lastName ?? '',
         phoneNumber: user?.contact?.phones?.[0] ?? '',
-        loginEmail: user?.loginEmail,
-    });
+    };
+
+    const [userDetailsFormData, setUserDetailsFormData] = useState(initialUserDetailsFormData);
+    const [discardConfirmationOpen, setDiscardConfirmationOpen] = useState(false);
 
     const onDiscardChangesClick = () => {
         setDiscardConfirmationOpen(true);
     };
 
     const handleDiscardChanges = () => {
-        discardChanges();
+        setUserDetailsFormData(initialUserDetailsFormData);
         setDiscardConfirmationOpen(false);
     };
 
+    const navigation = useNavigation();
+
+    const isUpdatingUserDetails =
+        navigation.state === 'submitting' &&
+        navigation.formAction === '/members-area/my-account/update-details';
+
+    const isResettingPassword =
+        navigation.state === 'submitting' &&
+        navigation.formAction === '/members-area/my-account/reset-password';
```

Comment:
I don't like this full route path here. seems too brittle.
far from malleable programming experience. In the small.